frontend/src/components/Registration.js

```diff
@@ -0,0 +1,135 @@
+import React, { Component } from 'react';
+import './registration.css';
+// import render from 'react-dom';
+// import { useHistory } from 'react-router-dom';
+import { withRouter } from 'react-router-dom';
+
+
+class Registration extends Component {
+
+    constructor(props) {
+        super(props);
+        // console.log(props)
+        this.state = {
+            newUser: {
+              username: '',
+              password: '',
+              kingdomname: ''
+            }
+          };

```

Comment:
@zvorsi this still doesnt have to be a separate newUser property on the state:
```suggestion
        this.state = {
              username: '',
              password: '',
              kingdomname: ''
          };

```